src/core/qgsproject.h

```diff
@@ -1631,7 +1631,7 @@ class CORE_EXPORT QgsProject : public QObject, public QgsExpressionContextGenera
      * \see layerWillBeRemoved()
      * \see layersRemoved()
      */
-    void layersWillBeRemoved( const QList<QgsMapLayer *> &layers );
+    void layersWillBeRemoved( const QList<QgsMapLayer *> &layers ) SIP_PYNAME( layersWillRemovedLayers );
```

Comment:
The proposed names miss a `Be`
What about just giving them a completely different name (and adapting some api docs to guide people to these signals).

```suggestion
    void layersWillBeRemoved( const QList<QgsMapLayer *> &layers ) SIP_PYNAME( aboutToRemoveLayers );
```

Reply:
in such case, we also change the cpp names, right?
Advantages:
* we can specifically write the since annotation for PyQGIS
* that avoids the overload in cpp

Reply:
Why not.
I'd like to check with @nyalldawson what he thinks about this API change

Reply:
Yeah I'm +1 to rename the cpp method too -- overloaded signals are just a pain.